### DISCUSSION TITLE
Information for building an Andino robot

### DISCUSSION THREAD

**Discussion by RomainMaure** [Ekumen-OS/andino#183] [Q&A] [answered]
Hello!

I would be interested in building an Andino and I have the following questions:
- I am not sure to understand the use of the Arduino Nano. Is it not possible to control the motor driver and read the encoder values using the GPIO pins of the Raspberry Pi?
- Could you provide a link for the power bank that you used?
- Could you provide more information with respect to the bolts and spacers that you used? In particular, what is the length of your spacers?

Best regards.

**Comment by francocipollone** [comment, marked as answer]
> Hello!
> 
> I would be interested in building an Andino and I have the following questions:
> 
> * I am not sure to understand the use of the Arduino Nano. Is it not possible to control the motor driver and read the encoder values using the GPIO pins or the Raspberry Pi?
> * Could you provide a link for the power bank that you used?
> * Could you provide more information with respect to the bolts and spacers that you used? In particular, what is the length of your spacers?
> 
> Best regards.

Hey there @RomainMaure, thanks for having interest on building an Andino :rocket: 

> I am not sure to understand the use of the Arduino Nano. Is it not possible to control the motor driver and read the encoder values using the GPIO pins or the Raspberry Pi?

The idea of the arduino nano/uno is to separate a bit the low level hardware from the high level hardware( SBC(raspi)). 

This way, the arduino nano's only job is to interact with the motor driver and offer commands via serial port. On top of that (in the raspi) a hardware_interface (ros2 control) is implemented to separate ros/controllers/high level soft from the low level hardware 
More info at https://github.com/Ekumen-OS/andino/tree/humble/andino_base
(Worth noting that support to IMU (optional) is about to be added, and this is also connected to the arduino).


Also note that once that you set up your arduino, you can use any other SBC instead of RasperryPi if you want. (Jetson Nano?, a laptop?).
And that's possible because there is a decoupling between low and high level hardware. 

Finally, Andino was thought for educational purposes, to be able to experiment with robotics quickly, without spending too much time building it, therefore a verified way of building it was provided. The fact that this is all public is because anyone can choose to take different path when building it if that is something that suits them.


> * Could you provide a link for the power bank that you used?

In general any 5V3A with 10000mAh powerbank works well. It is preffered to use some slim model to better fix the powerbank 
For the one I have I used this one: https://www.amazon.eg/-/en/Xiaomi-VXN4274GL-Charge-Power-10000mAh/dp/B08426JMHK 
but any other with similar characteristics would be ok.

If the powerbank is slim enough you can attach it at this location which is useful to save some space.
<img src=https://github.com/Ekumen-OS/andino/assets/53065142/7e7a3ae5-1588-4c6a-868f-d18d42a27e06 width=400 />



> Could you provide more information with respect to the bolts and spacers that you used? In particular, what is the length of your spacers?

Sure, typically with M3 bolts and nuts you will be ok for all the fixings. The raspberry pi will need M2 or M2.5 bolts.
Regarding the spacers with M3 50mm Male-Female spacers you will be ok. And for the Lidar using spacers of 30mm length is recommended.

We plan to improve the Bill of Materials section (https://github.com/Ekumen-OS/andino/tree/humble/andino_hardware)


### Finally

We are about to set up a Discord server for the community to interact and to provide support if needed. Stay tuned for that :smile:

**Reply by RomainMaure** [reply]
Thank you for your fast, clear and detailed answer :)

**Comment by Ahtisham123Ahmed** [comment]
hyy man did you use arduno ros bridge?
can you help me out. i am using ros2 humble

**Reply by francocipollone** [reply]
Hey there @Ahtisham123Ahmed , for the arduino - raspberry communication we are not relying on a ros bridge. Simply uart communication: As you can see in the andino_firmware package: https://github.com/Ekumen-OS/andino/blob/humble/andino_firmware/src/serial_stream_arduino.h

> can you help me out. i am using ros2 humble

Comment here or in the discussions sections what your doubts are and how we can help you
